packages/vite/src/node/optimizer/index.ts

```diff
@@ -1160,10 +1152,11 @@ const lockfileFormats = [
 })
 const lockfileNames = lockfileFormats.map((l) => l.name)
 
-function getConfigHash(config: ResolvedConfig, ssr: boolean): string {
+function getConfigHash(environment: Environment): string {
```

Comment:
I noticed that `getConfigHash` became unstable for my use case https://github.com/hi-ogawa/vite-plugins/pull/634 due to `environment.config.resolve.noExternal` included as a hash seed, which wasn't the case when `config.ssr.noExternal`.

I'm using https://github.com/svitejs/vitefu's `crawlFrameworkPkgs` to collect `noExternal` and it's not deterministic. I can normalize it on my side with `unique + sort`, but maybe we should remove it from `JSON.stringify` if `resolve.noExternal`  doesn't affect deps optimization?

Reply:
Maybe we should sort it ourselves then? 🤔 We sort other options

Reply:
Interesting. This is SSR only (we need to review if we can use them for the client too). And ya, I think it should not affect the fixed optimized deps we have during SSR 🤔
Maybe we should just remove it as you propose it for now.